package.json

```diff
@@ -106,33 +106,33 @@
-    "@opentelemetry/instrumentation-dns": "~0.25.0",
-    "@opentelemetry/instrumentation-express": "~0.25.0",
-    "@opentelemetry/instrumentation-graphql": "~0.25.0",
-    "@opentelemetry/instrumentation-grpc": "~0.24.0",
-    "@opentelemetry/instrumentation-hapi": "~0.25.0",
-    "@opentelemetry/instrumentation-http": "~0.24.0",
-    "@opentelemetry/instrumentation-ioredis": "~0.25.0",
-    "@opentelemetry/instrumentation-knex": "~0.25.0",
-    "@opentelemetry/instrumentation-koa": "~0.25.0",
-    "@opentelemetry/instrumentation-memcached": "~0.25.0",
-    "@opentelemetry/instrumentation-mongodb": "~0.25.0",
-    "@opentelemetry/instrumentation-mysql": "~0.25.0",
-    "@opentelemetry/instrumentation-mysql2": "~0.25.0",
-    "@opentelemetry/instrumentation-net": "~0.25.0",
-    "@opentelemetry/instrumentation-pg": "~0.25.0",
-    "@opentelemetry/instrumentation-pino": "~0.25.0",
-    "@opentelemetry/instrumentation-redis": "~0.25.0",
-    "@opentelemetry/instrumentation-winston": "~0.25.0",
-    "opentelemetry-instrumentation-amqplib": "~0.25.0",
-    "opentelemetry-instrumentation-aws-sdk": "~0.25.0",
-    "opentelemetry-instrumentation-elasticsearch": "~0.25.0",
-    "opentelemetry-instrumentation-kafkajs": "~0.25.0",
-    "opentelemetry-instrumentation-mongoose": "~0.25.0",
-    "opentelemetry-instrumentation-sequelize": "~0.25.0",
-    "opentelemetry-instrumentation-typeorm": "~0.25.0"
+    "@opentelemetry/instrumentation-bunyan": "~0.24.0",
+    "@opentelemetry/instrumentation-cassandra-driver": "~0.23.0",
+    "@opentelemetry/instrumentation-dns": "~0.23.0",
+    "@opentelemetry/instrumentation-express": "~0.23.0",
+    "@opentelemetry/instrumentation-graphql": "~0.23.0",
+    "@opentelemetry/instrumentation-grpc": "~0.23.0",
+    "@opentelemetry/instrumentation-hapi": "~0.23.0",
+    "@opentelemetry/instrumentation-http": "~0.23.0",
+    "@opentelemetry/instrumentation-ioredis": "~0.23.0",
+    "@opentelemetry/instrumentation-knex": "~0.23.0",
+    "@opentelemetry/instrumentation-koa": "~0.23.0",
+    "@opentelemetry/instrumentation-memcached": "~0.23.0",
+    "@opentelemetry/instrumentation-mongodb": "~0.23.0",
+    "@opentelemetry/instrumentation-mysql": "~0.23.0",
+    "@opentelemetry/instrumentation-mysql2": "~0.23.0",
+    "@opentelemetry/instrumentation-net": "~0.23.0",
+    "@opentelemetry/instrumentation-pg": "~0.23.0",
+    "@opentelemetry/instrumentation-pino": "~0.23.0",
+    "@opentelemetry/instrumentation-redis": "~0.23.0",
+    "@opentelemetry/instrumentation-winston": "~0.24.0",
+    "opentelemetry-instrumentation-amqplib": "~0.23.0",
+    "opentelemetry-instrumentation-aws-sdk": "~0.23.0",
+    "opentelemetry-instrumentation-elasticsearch": "~0.23.0",
+    "opentelemetry-instrumentation-kafkajs": "~0.23.0",
+    "opentelemetry-instrumentation-mongoose": "~0.23.0",
+    "opentelemetry-instrumentation-sequelize": "~0.23.0",
+    "opentelemetry-instrumentation-typeorm": "~0.23.0"
```

Comment:
Why were these downgraded? :eyes:

Reply:
Because when we install the local version we'd have conflicts in the examples. The peer dep versions are very restrictive right now.

The current solution is to update peer deps with the main package and used version in the examples.
An alternative solution would be to detect the local unpublished versions for peer deps and install those with the local compiled version of the distro package itself:

1. compile distro
2. read peer dep versions from package.json
3. per example do:
    1. see the new peer dep versions
    2. `npm install splunk-otel-js.tgz @opentelemetry/instrumentation-package@newversion`.

Reply:
I'd be happy do implement the described solution in another PR.

Reply:
But won't the downgraded peer deps cause issues for users? I.e. they'd do `npm install @opentelemetry/instrumentation-x`, get the 0.25.0 one and then get a bunch of warnings due to peer dep versions?
Could we perhaps start pinning them to something like `^0.23.0`?

Reply:
Yes, maybe.
`^0.23.0` is the same as `~0.23.0` in 0.x, but I get what you mean - make it looser. I did think about it - but "how loose?" is the question.

From one side we'd practically want to have `* excluding anything that might break us` because, why not - instrumentations are a way to the fine control over the SDK.
On the other hand, we'd want to control the variability of the bundle in the wild to avoid solving issues with compatibility etc. In that extreme we would not care about people not being able to just npm install the latest, because we've got a list of instrumentation-version tuples we support and if they need others, they should upgrade to the later version of our package.

Thinking through all this during the development of this PR I concluded that it's outside of the scope for now. I'm turning back an unreleased change in this PR just to be able to go through that again explicitly with having the above in mind.

**My suggestion is still to leave it as is in this PR but, before the next release upgrade all instrumentation packages back to their latest versions, but hold them locked**. We can always loosen them once we get customer feedback, but it's more difficult to go the other way.

Reply:
But wouldn't it best to have an up to date version that works with the latest OTel? Should get easier once 1.0 is reached, I'm just afraid we might make it too annoying for users to install anything. :shrug:

Reply:
And the point about relaxing peer deps is fine, it's that we'd have an upper limit on the versions we are compatible with and in an ideal world we are always (as in same day) compatible with the latest OTel versions, unless they do a major breaking change.